### PR TITLE
feat(fromOpenApi): add browser support

### LINKED ISSUE #49: OpenAPI: Browser support without polyfills
## Description

The function `fromOpenApi` currently doesn't work on the browser without a lot of Node.js polyfills.

- The file [open-api-utils.ts](https://github.com/mswjs/source/blob/main/src/open-api/utils/open-api-utils.ts) uses status codes from `node:http`.
- The transitive dependency @apidevtools/swagger-parser > [@apidevtools/json-schema-ref-parser](https://github.dev/APIDevTools/json-schema-ref-parser) uses the `path`, `fs`, `http` and `https` modules. It also references `process.cwd`.

## Solution

For [open-api-utils.ts](https://github.com/mswjs/source/blob/main/src/open-api/utils/open-api-utils.ts), inlining the `STATUS_CODE` variable is easy enough.

For [@apidevtools/json-schema-ref-parser](https://github.dev/APIDevTools/json-schema-ref-parser), since it is the ubiquitous library used for JSON Schema dereferencing in Node, it may not make sense to replace it in Node. We could create a separate package entrypoint for the browser that prebundles the Node.js modules or replaces the library with web-compatible alternative (that may not support all features of JSON Schema dereferencing, but works for the web.)

## Alternatives

- Ask consumers of the library to bring their own polyfills. Will hurt the ability of teams to adopt this library. (I haven't been able to fully polyfill everything with Webpack, so I don't even know if it's possible.)
- Make @mswjs/source/open-api a Node.js only feature.
- Raise this issue to the `@apidevtools` folks. Make the change there.

### PR DESCRIPTION
**Pull request by tjosepo**
Closes #49

## Changes 
- Replaces `import { STATUS_CODES } from 'node:http'` with `import { STATUS_CODES } from './status-codes.js'`, this way, we don't depend on Node internals.
- Replaces `@apidevtools/swagger-parser` with a custom JSON Schema dereferencing function that works on the browser
- Adds [@stoplight/json](https://www.npmjs.com/package/@stoplight/json) to the project, to help with dereferencing JSON schemas.
  - I tried using [@stoplight/http-spec](https://www.npmjs.com/package/@stoplight/http-spec) as was suggested by @weyert, but it had issues with nested references, required a lot more code change, and was generally lacking in documentation. Using `@stoplight/json` directly was simpler. With a bit of recursion, the dereferencing algorithm really wasn't that bad.
- All tests are green ✅

> [!CAUTION]
> The custom dereferencing function doesn't support **remote references**. 
>
> Only local references are currently supported. I think it's best this way, since figuring out how to support remote references in a way that works on both Node.js and the browser will require more time. Restricting the input while we figure it out seems best.

### CODE REVIEW

File: src/open-api/from-open-api.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const supportedHttpMethods = Object.keys(
 export async function fromOpenApi(
   document: string | OpenAPI.Document | OpenAPIV3.Document | OpenAPIV2.Document,
```
Comment (tjosepo) [inline src/open-api/from-open-api.ts:22]: Just noticed the `string` type here.

What should happen when we pass in a `string`? Should we support `yaml` or only `json` format?

Currently, my dereference function only supports `objects`.

Comment (weyert) [reply src/open-api/from-open-api.ts:22]: Thank you for your pull request :)

Yeah, good point. I think we probably want to support both `yaml` and `json`? I thought one of those `yaml` packages support both `json` and `yaml` but I could be imagining that.

Comment (tjosepo) [reply src/open-api/from-open-api.ts:23]: You're right! The [yaml](https://www.npmjs.com/package/yaml) package support both JSON and YAML syntax, so we don't need to handle that logic ourselves!

I added the library and some test cases.

File: src/open-api/utils/dereference.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { pointerToPath } from '@stoplight/json'
+
+/**
+ * TODO: Support remote references.
```
Comment (weyert) [inline src/open-api/utils/dereference.ts:4]: I think this is fine for a first iteration. Remote reference can also easily open a can of worms, e.g. from which relative path would it load the remote files etc?

**Review by weyert — APPROVED** [review]
I think it's a great start to get it working in the browser. Only it doesn't appear to work when using `@vitest/browser` for me but appears to be related to `mswjs` itself. Need to play around a bit with that.

**Comment by weyert** [comment]
> >I think it's a great start to get it working in the browser. Only it doesn't appear to work when using @vitest/browser for me but appears to be related to mswjs itself. Need to play around a bit with that.
> 
> 
> 
> I tested my changes on Chrome using a small OpenAPI spec. It worked. Didn't try it with `@vitest/browser`.

I think this because in Vitest you typically try to use the Node version instead of the browser version needed for `@vitest/browser`. At least for me it's try to look for mswjs/node

**Comment by kettanaito** [comment]
I've added a simple browser test using `@vitest/browser`. It passes locally 🎉

**Review by kettanaito — APPROVED** [review]
This is absolutely incredible! Thank you for your work on this, @tjosepo 👏

**Comment by weyert** [comment]
Awesome great work! 🚀

**Comment by kettanaito** [comment]
## Released: v0.3.0 🎉

This has been released in v0.3.0!

- 📄 [**Release notes**](https://github.com/mswjs/source/releases/tag/v0.3.0)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/source/v/0.3.0)

Make sure to always update to the latest version (`npm i @mswjs/source@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**